Contributing-code-to-Oppia.md

```diff
@@ -1,307 +1,211 @@
+* **Want help figuring out what to do?** Just ask us on [Gitter](https://gitter.im/oppia/oppia-chat), or send an email to admin@oppia.org. We'll try to help!
 
-   - Once you've addressed everything, and would like the reviewer(s) to take another look:
-     - Follow the instructions in Step 3 to test your changes locally before pushing.
-     - Make the push, and then immediately check that the changes in the "Files Changed" tab are what you intend them to be.
-     - **Important:** Make sure you've posted responses to **all** the review comments from the reviewer(s).
-     - **Important:** In the conversation thread, **write a top-level comment** explicitly asking the reviewer(s) to take another look ("@XXX PTAL"), and assign them to the PR. If however, you forget to assign the reviewer(s), Oppiabot will assign them for you.
+If an issue hasn't got someone assigned to it, and there's no existing PR for the issue (you can check this by scanning the list of [existing PRs](https://github.com/oppia/oppia/pulls)), feel free to take it up by assigning yourself to it. You don't need to ask permission to do so. Also, if you need help or advice on an issue, you can contact the corresponding team lead, whose GitHub username you can find in the description of the project the issue is assigned to.  You can also find a list of all Oppia teams on the [Projects page](https://github.com/oppia/oppia/projects).
 
-7. **Make sure all CI checks pass.** While waiting to get LGTM from reviewers, make sure that all the CI checks (Travis, CircleCI, etc.) pass, since otherwise you won't be able to merge your PR. (See "[If your build fails...](https://github.com/oppia/oppia/wiki/If-your-build-fails)" for some suggestions on what to do if you run into issues.)
+## Tips for Success
 
-   - If all reviewers have given LGTM but you're still waiting for the CI checks to pass, make sure you're assigned to the PR, so that you can merge it once the CI checks are complete.
+1. Make dependent (“stacked”) PRs to ensure smaller time between reviews and subsequent PRs. A large PR results in difficulty to review for the reviewer as well as difficulty in making changes according to a review for an author. So, it is better to create smaller PRs which deliver a single small goal. If you have any other code change dependent on a PR, it is better to stack it on the that PR and create a new PR instead of merging it in one PR.
 
-8. **Tidy up!** After the PR status has changed to "Merged", delete the feature branch from both your local clone and the GitHub repository:
+1. Try and follow test-driven development. This is the easiest way to make sure the code you wrote is working fine. Writing tests for the expected behaviour, and then writing code that will pass the tests is the basic idea. Refer our guides for writing good tests:
```

Comment:
Add comma after "will pass the tests" (or restructure the sentence so there's no ambiguity on how to parse it)

Reply:
done